pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,95 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		return s.setControllerRef(keystoreSecret)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      i.GetTruststoreSecretName(),
+			Namespace: i.Namespace,
+		},
+	}
+
+	if trustSecret.Name == "" {
+		// It's not possible for client cert to be configured if no truststore secret is provided by the user
+		// as no certificates are available to be added to the server's truststore, resulting in no client's
+		// being able to authenticate with the server.
+		return &reconcile.Result{}, fmt.Errorf("Field 'ClientCertSecretName' must be provided for '%s' or '%s' to be configured",
+			ispnv1.ClientCertAuthenticate, ispnv1.ClientCertValidate)
+	}
+
+	if result, err := kube.LookupResource(trustSecret.Name, i.Namespace, trustSecret, s.client, s.log); result != nil {
```

Comment:
We also need to separate timeout for auto-generated and customer provided resources in `LookupResource` as for customer actions it could be too fast. Probably not in this PR

Reply:
+1 to another PR, there's a lot going on here already.